api/api.py

```diff
@@ -0,0 +1,510 @@
+    db: Session = Depends(get_db),
+):
+    """API endpoint to get participant names, lawyer and law firm with optional
+    search and limit parameters"""
+    query = db.query(ParticipantAssignment).options(
+        joinedload(ParticipantAssignment.participant),
+        joinedload(ParticipantAssignment.lawyer).joinedload(Lawyer.law_firm),
+    )
+
+    if limit is not None:
+        query = query.limit(limit)
+
+    if participant is not None:
+        query = query.join(ParticipantAssignment.participant).where(
+            Participant.participant_name.ilike(f"%{participant}%")
+        )
+    if lawyer is not None:
+        query = query.join(ParticipantAssignment.lawyer).where(
+            Lawyer.lawyer_name.ilike(f"%{lawyer}%")
+        )
+    if law_firm is not None:
+        query = (
+            query.join(ParticipantAssignment.lawyer)
+            .join(Lawyer.law_firm)
+            .where(LawFirm.law_firm_name.ilike(f"%{law_firm}%"))
+        )
+
+    result = query.all()
+    return result
+
+
+@app.get("/tags/", response_model=List[TagModel])
+def read_tags(
+    limit: Optional[int] = Query(None, description="Limit the number of results"),
+    search: Optional[str] = Query(None, description="Tag name to filter by"),
+    db: Session = Depends(get_db),
+):
+    """API endpoint to get tag names with optional search and limit parameters"""
+    query = db.query(Tag)
+    if search is not None:
+        query = query.where(Tag.tag_name.ilike(f"%{search}%"))
+    if limit is not None:
+        query = query.limit(limit)
+    return query.all()
+
+
+@app.get("/verdicts/", response_model=List[VerdictModel])
+def read_verdicts(db: Session = Depends(get_db)):
+    """API endpoint to get verdicts"""
+    query = db.query(Verdict)
+    return query.all()
+
+
+@app.get("/court_cases/", response_model=List[CourtCaseModel])
+def read_court_cases(
+    tag: Optional[str] = Query(None, description="Tag name to filter by"),
+    judge: Optional[str] = Query(None, description="Judge name to filter by"),
+    participant: Optional[str] = Query(
+        None, description="Participant name to filter by"
+    ),
+    lawyer: Optional[str] = Query(None, description="Lawyer name to filter by"),
+    law_firm: Optional[str] = Query(None, description="Law firm name to filter by"),
+    title: Optional[str] = Query(None, description="Title to filter by"),
+    citation: Optional[str] = Query(None, description="Citation to filter by"),
+    verdict: Optional[str] = Query(None, description="Verdict to filter by"),
+    court: Optional[str] = Query(None, description="Court name to filter by"),
+    start_date: Optional[date] = Query(
+        None, description="Start date to filter by in the form YYYY-MM-DD"
+    ),
+    end_date: Optional[date] = Query(
+        None, description="End date to filter by in the form YYYY-MM-DD"
+    ),
+    limit: Optional[int] = Query(None, description="Limit the number of results"),
+    db: Session = Depends(get_db),
```

Comment:
I may set a default limit for the cases to avoi it taking too long and also because I don't think nayone would need to see all cases at once

Reply:
Hi Anna, 

This is a really great suggestion. This has been implemented in commit [0e8cc3d](https://github.com/LinfanS/court-transcripts-pipeline/pull/17/commits/0e8cc3d0befbcab20d030f5e7961049ed9e92526), now we are setting a default limit of 100 results, while still supporting the option to return all results using a value of -1 for the limit. This will help with performance particularly the `/court_cases/` endpoint. Thanks.